### PR TITLE
VZ-11699: Refactor Analysis Tool Code To Remove Unclear Function Names 

### PR DESCRIPTION
**Pull request by George-Aeillo**
This PR changes some of the function names in the analysis tool to more accurately describe how they behave. This involved refactoring the analysis tool code and associated unit tests.

### CODE REVIEW

File: tools/vz/pkg/analysis/internal/util/files/files_test.go [hunk 1]
```diff
@@ -13,39 +13,39 @@ import (
 	"go.uber.org/zap"
 )
 
-// TestGetMatchingFilesGood Tests that we can find the expected set of files with a matching expression
+// TestGetMatchingFileNamesGood Tests that we can find the expected set of files with a matching expression
 // GIVEN a call to GetMatchingDirectories
```
Comment (marcogelle) [inline tools/vz/pkg/analysis/internal/util/files/files_test.go:17]: This GIVEN comment should be changed to GetMatchingFileNames

File: tools/vz/pkg/analysis/internal/util/files/files_test.go [hunk 2]
```diff
@@ -13,39 +13,39 @@ import (
...
 	assert.Nil(t, myFiles)
 }
 
-// TestGetMatchingDirectoriesGood Tests that we can find the expected set of files with a matching expression
-// GIVEN a call to GetMatchingDirectories
+// TestGetMatchingDirectoryNamesGood Tests that we can find the expected set of files with a matching expression
+// GIVEN a call to GetMatchingDirectoryNames
 // WHEN with a valid rootDirectory and regular expression
 // THEN files that matched will be returned
 func TestGetMatchingDirectoriesGood(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (George-Aeillo) [reply tools/vz/pkg/analysis/internal/util/files/files_test.go:38]: Ok! Thanks!

File: tools/vz/pkg/analysis/internal/util/files/files_test.go [hunk 3]
```diff
@@ -56,14 +56,14 @@ func TestGetMatchingDirectoriesGood(t *testing.T) {
 // THEN we get failures as expected
 func TestGetMatchingInvalidInputs(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (George-Aeillo) [reply tools/vz/pkg/analysis/internal/util/files/files_test.go:57]: Ok! Thanks!

File: tools/vz/pkg/analysis/internal/util/files/files_test.go (hunk 1 above)
Comment (George-Aeillo) [reply tools/vz/pkg/analysis/internal/util/files/files_test.go]: Ok! Thanks!

**Review by marcogelle — APPROVED** [review]

**Pull request merged**